### PR TITLE
Remove dependencyManagement in shaded zookeeper pom

### PR DESCRIPTION
**Pull request by pan3793**
### _Why are the changes needed?_

To gain a cleaner `pom.xml`.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [x] Add screenshots for manual tests if appropriate

Manually tested, and the `dependencyManagement` block is removed after this change.
```
  ...
  <dependencyManagement>
    <dependencies>
      <dependency>
        <groupId>org.apache.zookeeper</groupId>
        <artifactId>zookeeper</artifactId>
        <version>${zookeeper.version}</version>
      </dependency>
      <dependency>
        <groupId>io.netty</groupId>
        <artifactId>netty-bom</artifactId>
        <version>${netty.version}</version>
        <type>pom</type>
        <scope>import</scope>
      </dependency>
    </dependencies>
  </dependencyManagement>
  ...
```

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by pan3793** [comment]
cc @zhouyifan279

File: kyuubi-relocated-zookeeper-parent/kyuubi-relocated-zookeeper-36/pom.xml [hunk 1]
```diff
@@ -38,23 +38,6 @@ under the License.
...
 
-    <dependencyManagement>
-        <dependencies>
-            <dependency>
-                <groupId>org.apache.zookeeper</groupId>
-                <artifactId>zookeeper</artifactId>
-                <version>${zookeeper.version}</version>
-            </dependency>
-            <dependency>
-                <groupId>io.netty</groupId>
```
Comment (zhouyifan279) [inline kyuubi-relocated-zookeeper-parent/kyuubi-relocated-zookeeper-36/pom.xml:49]: Just be curious, why netty-bom was needed before ?

Comment (pan3793) [reply kyuubi-relocated-zookeeper-parent/kyuubi-relocated-zookeeper-36/pom.xml:49]: netty is the transitive deps of Zookeeper

**Comment by pan3793** [comment]
Merged to master
